### ISSUE TITLE
Fresh install of MSYS2 emits error due to cygbaseheap

### ISSUE THREAD

**Issue by ToxiClay** [msys2/MINGW-packages#7679]
While running `pacman -Syyu` on a fresh install of MSYS2 onto a Windows 7 virtual machine (binary installer obtained [here](https://www.msys2.org/)), I encounter the following error:

```
0 [main] pacman (1988) C:\dev\msys64\usr\bin\pacman.exe: *** fatal error - cygheap base mismatch detected - 0x18034B408/0x180345408.
This problem is probably due to using incompatible versions of the cygwin DLL.
Search for cygwin1.dll using the Windows Start->Find/Search facility
and delete all but the most recent version.  The most recent version *should*
reside in x:\cygwin\bin, where 'x' is the drive on which you have
installed the cygwin distribution.  Rebooting is also suggested if you
are unable to find another cygwin DLL.
```

This error is non-fatal; MSYS2 continues processing and I am able to do my work, but I am slightly confused because I don't have a copy of Cygwin at all. At first, I believed the problem to be due to the fact that I had installed Git for Windows first, which apparently comes with a mingw unit environment (per evandrocoan [here](https://github.com/actions/virtual-environments/issues/594)), but I was able to replicate this issue even without Git for Windows installed.

Is this a known issue, and is there a solution planned?

**Comment by jeremyd2019** [comment]
msys2/MSYS2-packages#2295

MSYS2 contains a fork of Cygwin, but it is called msys-2.0.dll.  Git for Windows contains a patched version of MSYS2

**Comment by ToxiClay** [comment]
> MSYS2 contains a fork of Cygwin, but it is called msys-2.0.dll. Git for Windows contains a patched version of MSYS2

I understand this, but please note that the problem occurred even when I did not install Git for Windows.

**Comment by revelator** [comment]
something might have locked the msys-2.0.dll so it failed updating (some antivirus do nasty things with cygwin / mingw* based tools). Best approach would be turning of protection then get the latest msys2-runtime and copy the msys-2.0.dll to c:\msys2\usr\bin overwriting the old one. See if it fixes things and report back.

**Comment by ToxiClay** [comment]
But the problem isn't with msys-2.0.dll, though. It's with cygwin1.dll.

Steps taken:

1. Reset VM to pre-MSYS2 baseline.
2. Ensured Windows was not being monitored.
3. Installed MSYS2.
4. Ran `pacman -Syyu`.
5. Encountered error.

There's no time here for me to replace msys-2.0 even if this were the problem.

Curiously, though, my computer doesn't have `cygwin1.dll` on it at all. See screenshot: 
![image](https://user-images.githubusercontent.com/13796683/105797635-5d312b00-5f5e-11eb-9d2b-57d07196c100.png)

So now I'm even less sure what MSYS is complaining about.

**Comment by revelator** [comment]
Hmm might just be an overlooked printf complaining about the cygwin dll since msys-2.0 is more or less based based on the cygwin source code.

You should however still be able to open the package with the msys-2.0 runtime with 7zip and copy the dll to /usr/bin overwriting the old one (dont use the msys2 shell while doing that).

**Comment by jeremyd2019** [comment]
> Hmm might just be an overlooked printf complaining about the cygwin dll since msys-2.0 is more or less based based on the cygwin source code.
Correct.  It is a message coming from msys-2.0.dll, which says cygwin1.dll

> You should however still be able to open the package with the msys-2.0 runtime with 7zip and copy the dll to /usr/bin overwriting the old one (dont use the msys2 shell while doing that).

Unnecessary.  This error is due to the current state of affairs updating from the current installer: there are updates to both msys2-runtime and bash, and bash is triggering a pacman (libalpm) hook to update info dirs.  That process is failing with the "fatal" error message, because the msys-2.0.dll has been updated without all processes using it being restarted (namely, pacman itself, and the shell it was launched from, etc).  It is not "fatal" to the entire update or the installation, only updating the info directories, and can be safely ignored for the most part (with the possible exception of if you actually intend to use the info pages, in which case you might want to regenerate the info directories after restarting all msys2 processes).

**Comment by revelator** [comment]
Ok then :)

**Comment by liudonghua123** [comment]
I have the same issue, my git and msys64 both have `msys-2.0.dll` with different version.

![image](https://user-images.githubusercontent.com/2276718/148328995-58480611-3652-47d3-b65b-8e75f69cd962.png)

If in git bash I use some commands in msys2, then the error occurred.

![image](https://user-images.githubusercontent.com/2276718/148329410-21877307-b436-4951-81d5-61f6e98146f0.png)
